pkg/webhooks/machine_webhook_test.go

```diff
@@ -2382,6 +2382,14 @@ func TestValidateAWSProviderSpec(t *testing.T) {
 			expectedOk:    false,
 			expectedError: "providerSpec.userDataSecret: Required value: expected providerSpec.userDataSecret to be populated",
 		},
+		{
+			testCase: "with no user data secret name it fails",
+			modifySpec: func(p *machinev1beta1.AWSMachineProviderConfig) {
+				p.UserDataSecret.Name = ""
+			},
+			expectedOk:    false,
+			expectedError: "providerSpec.userDataSecret.name: Required value: expected providerSpec.userDataSecret.name name must be provided",
```

Comment:
```suggestion
			expectedError: "providerSpec.userDataSecret.name: Required value: expected providerSpec.userDataSecret.name to be provided",
```

Reply:
thanks @damdo I had tested locally that the error appears and I have tried to match in the unit test:

error: machinesets.machine.openshift.io "gparente0808aws-nnvg2-worker-eu-west-2a" could not be patched: admission webhook "validation.machineset.machine.openshift.io" denied the request: providerSpec.userDataSecret.name: Required value: name must be provided

Unfortunately, I have not managed to run go test locally. I will fix asap